app/workers/hg/message_worker.rb

```diff
@@ -58,20 +69,46 @@ def perform(user_id, redis_namespace, bot_class_name)
                 long:  attachment['payload']['coordinates']['long']
               }
             )
+
+            # Set location attachment chatbase fields
+            if ENV['CHATBASE_API_KEY']
+              set_chatbase_fields(request.intent, message.text, false)
+            end
+
           else
             # TODO: What should we do if attachments aren't recognized?
           end
         # If the user is in the middle of a dialog...
         elsif user.context[:dialog_action]
           request = build_dialog_request(user, message)
 
-          # If the message is text...
+          # Set chatbase fields for dialog action
+          if ENV['CHATBASE_API_KEY']
+            set_chatbase_fields(request.action, message.text, false)
+          end
+
+        # If the message is text...
         else
           # Parse the message.
           nlu_response, params = parse_message(message.text, user)
 
           # Build a request.
           request = build_request(message, nlu_response, params, user)
+          
+          # Set chatbase fields and determine not_handled
+          if ENV['CHATBASE_API_KEY']
```

Comment:
To avoid another block of nesting, and, you could also do the following:

```ruby
return unless ENV['CHATBASE_API_KEY']
```

You can often avoid conditional blocks with these pre-emptive return clauses at the beginning of a method.

Reply:
I just realized this wasn't at the end of the method, but rather inside of a conditional block. So ignore that advice for this particular bit. :)